tests.c

```diff
@@ -158,6 +159,224 @@ test_opcode_0x13(void) // NOLINT
   CU_ASSERT(cpu.d == 1);
 }
 
+void
+test_opcode_0x05(void) // NOLINT
+{
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu.b = 0xF;                                      // NOLINT
+  int code_found = execute_instruction(&cpu, 0x05); // NOLINT
+
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT_EQUAL(cpu.b, 14);
+  CU_ASSERT((cpu.flags & FLAG_Z) == 0);
+  CU_ASSERT((cpu.flags & FLAG_S) == 0);
+  CU_ASSERT((cpu.flags & FLAG_P) == 0);
+  CU_ASSERT((cpu.flags & FLAG_AC) == FLAG_AC);
+}
+
+void
+test_opcode_0x0e(void)
+{
+  // MVI C, D8
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu.pc = 0x4341;                       // NOLINT
+  cpu_write_mem(&cpu, cpu.pc + 1, 0x19); // NOLINT
+
+  int code_found = execute_instruction(&cpu, 0x0e); // NOLINT
+
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT_EQUAL(cpu.c, 0x19); // NOLINT
+}
+
+void
+test_opcode_0x19(void) // NOLINT
+{
+  // DAD D
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu.d = 255; // NOLINT
+  cpu.h = 63;  // NOLINT
+  cpu.l = 200; // NOLINT
+
+  int code_found = execute_instruction(&cpu, 0x19); // NOLINT
+
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT(cpu.h == 64);
+  CU_ASSERT(cpu.l == 199);
+  CU_ASSERT((cpu.flags & FLAG_CY) == 0);
+}
+
+void
+test_opcode_0x26(void)
+{
+  // MVI H, D8
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu.pc = 0x4563;                       // NOLINT
+  cpu_write_mem(&cpu, cpu.pc + 1, 0x65); // NOLINT
+
+  int code_found = execute_instruction(&cpu, 0x26); // NOLINT
+
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT_EQUAL(cpu.h, 0x65); // NOLINT
+}
+
+void
+test_opcode_0x36(void)
+{
+  // MOV M, D8
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu.pc = 0x4352;                       // NOLINT
+  cpu.h = 0x23;                          // NOLINT
+  cpu.l = 0x12;                          // NOLINT
+  cpu_write_mem(&cpu, cpu.pc + 1, 0xA1); // NOLINT
+
+  int code_found = execute_instruction(&cpu, 0x36); // NOLINT
+
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT_EQUAL(cpu_read_mem(&cpu, 0x2312), 0xA1); // NOLINT
+}
+
+void
+test_opcode_0x5e(void)
+{
+  // MOV E, M
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu.h = 0x12;                      // NOLINT
+  cpu.l = 0x31;                      // NOLINT
+  cpu_write_mem(&cpu, 0x1231, 0x45); // NOLINT
+
+  int code_found = execute_instruction(&cpu, 0x5e); // NOLINT
+
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT_EQUAL(cpu.e, 0x45); // NOLINT
+}
+
+void
+test_opcode_0x7a(void) // NOLINT
+{
+  // MOV A,D
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu.d = 100; // NOLINT
+
+  int code_found = execute_instruction(&cpu, 0x7a); // NOLINT
+
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT(cpu.a == 100);
+}
+
+void
+test_opcode_0xa7(void) // NOLINT
+{
+  // ANA A
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  cpu.a = 85; // NOLINT
+  // S and CY flag set.
+  cpu.flags = 136; // NOLINT
```

Comment:
I assume this is all flags set? Maybe clearer to write it out in binary or something?